### ISSUE TITLE
Pretty regular crashes always in kernelbase.dll - log always shows memory leaks

### ISSUE THREAD

**Issue by lcalder** [obsproject/obs-studio#4230]
## Platform

Operating system and version: Windows 10
OBS Studio version: 26.0 through 26.1.1

## Expected Behavior
OBS shouldn't keep crashing with Unhandled exception: e06d7363
Fault address: 7FF92F0CD759 (c:\windows\system32\kernelbase.dll)kernelbase.dll

## Current Behavior

Crashes pretty regularly for last several months. I tried posting in other places on this site but I think that was the wrong place.

## Steps to Reproduce

Use OBS to switch between various scenes, at least one of which uses window capture. Please note, OBS may work for a few hours at a time before crashins and with many scene switches prior to crashing... Sometimes it will crash after a shorter period. 

I believe switching to/from a scene with window capture repeated will cause the crash much more quickly, and indeed, my "zoom" scene, which uses window capture, has a projector going always on screen even when I am in other scenes and I will suddenly notice the zoom scene has frozen, but the other scenes I am using will still work (they contain cameras and screen captures which haven't frozen yet, etc). Before when I was using window capture for MS Whiteboard, the crashes happened much more frequently, since it is a scene I switch to repeatedly, however, after switching that scene to use "screen capture" the frequency of crashes has diminished, as I am using fewer window captures... (maybe 4 or 5 in various scenes).
I ran OBS v25 for six months and never had a problem with window captures, so I think this problem has recently come about.

## Additional information

Never seems to crash in the same place. I have several scenes, one of which is a windows capture of a "zoom" window. One thing that 50% of time precedes a crash is the zoom scene projector freezes and my other scenes will still work for a few minutes. Once I notice it has frozen, I will try to exit, by first stopping the virtual camera, which succeeds, and then clicking on the X to exit, whereupon it hangs and must be killed. Alternatively it may just crash at that point and offer me a chance to copy the crash log to the clipboard. Interestingly , when it crashes with this message, when I try to relaunch OBS, I am told it is still running, and so I have to kill it in the task manager and then it will restart.

I have already systematically removed each of the plugins I was using, including Replay Source (which has some bugs of its own), Advanced Scene Switcher, some lua clock and date scripts. I haven't removed the NDI or websockets plugins, but can if suggested.

When I remove these plugins, the crashes still occur.

I have noticed fewer crashes if I use "screen capture" for most of my scenes instead of "window capture", which especially when capturing my Foxit PDF viewer, will almost always crash within a few minutes of switching between that scene and other scenes.
I now use "screen capture" for Microsoft Whiteboard, but before when I used window capture it crashed pretty regularly after upgrading to OBS version 26+. I remember using windows capture extensively on OBS version 25 with never a crash, but now, for some reason, when I try a portable install of OBS version 25, I can't get it to work without crashes, so maybe a bug was introduced with the last OBS 25 release and I didn't roll back far enough to an earlier version. I used OBS version 25 reliably for 6 months or more and never had these crashes, so it would sure be nice to find the problem...

[Crash 2021-02-12 15-44-55.txt](https://github.com/obsproject/obs-studio/files/5975170/Crash.2021-02-12.15-44-55.txt)
[Crash 2021-02-10 19-55-31.txt](https://github.com/obsproject/obs-studio/files/5975171/Crash.2021-02-10.19-55-31.txt)
[2021-02-12 20-36-19.txt](https://github.com/obsproject/obs-studio/files/5975173/2021-02-12.20-36-19.txt)
[2021-02-12 15-46-26.txt](https://github.com/obsproject/obs-studio/files/5975175/2021-02-12.15-46-26.txt)

I am happy to provide more logs and crash reports, as I have lots, or try any suggestions.

**Comment by WizardCM** [comment]
Please properly fill in the issue template. Additionally, this bug might already be fixed in the nightly. Try [this build](https://obsbot.rodney.io/artifacts/40812962).

**Comment by lcalder** [comment]
sorry I didn't see a template
thanks for the suggestion

**Comment by lcalder** [comment]
Platform

Operating system and version: Windows 10
OBS Studio version: 26.0 through 26.1.1
Expected Behavior

OBS shouldn't keep crashing with Unhandled exception: e06d7363
Fault address: 7FF92F0CD759 (c:\windows\system32\kernelbase.dll)kernelbase.dll

Current Behavior

Crashes pretty regularly for last several months. I tried posting in other places on this site but I think that was the wrong place.

Steps to Reproduce

    Use OBS to switch between various scenes, at least one of which uses window capture. Please note, OBS may work for a few hours at a time before crashins and with many scene switches prior to crashing... Sometimes it will crash after a shorter period. 
I believe switching to/from a scene with window capture repeated will cause the crash much more quickly, and indeed, my "zoom" scene, which uses window capture, has a projector going always on screen even when I am in other scenes and I will suddenly notice the zoom scene has frozen, but the other scenes I am using will still work (they contain cameras and screen captures which haven't frozen yet, etc). Before when I was using window capture for MS Whiteboard, the crashes happened much more frequently, since it is a scene I switch to repeatedly, however, after switching that scene to use "screen capture" the frequency of crashes has diminished, as I am using fewer window captures... (maybe 4 or 5 in various scenes).
I ran OBS v25 for six months and never had a problem with window captures, so I think this problem has recently come about.


Additional information

Please see logs and crashlogs in the initial report above.
Thanks for any help.

**Comment by dodgepong** [comment]
I have manually updated your OP with the issue template.

**Comment by lcalder** [comment]
Thanks so much. I copied from another post but it didn't format the same way. In future, do I format it manually using CSS or is the one that can be copied that preserves the formatting?

**Comment by Fenrirthviti** [comment]
When you open a new issue, the template is present in the post automatically. You would have to have deleted the text to avoid it being present. It uses github flavored markdown for formatting.

**Comment by lcalder** [comment]
happened again.
Crash file and Log file attached.
[2021-02-14 08-39-03.txt](https://github.com/obsproject/obs-studio/files/5979430/2021-02-14.08-39-03.txt)

[Crash 2021-02-14 12-11-17.txt](https://github.com/obsproject/obs-studio/files/5979429/Crash.2021-02-14.12-11-17.txt)

**Comment by WizardCM** [comment]
Have you tried the nightly?

**Comment by lcalder** [comment]
I will try right now.

**Comment by lcalder** [comment]
Yes, it crashed within 5 minutes... faster than normal... the Zoom window capture froze but I was still able to interact with the program enough to stop the recording and try to exit, whereupon it hung and had to be killed, but didn't dump core (or crashfile).

**Comment by lcalder** [comment]
Interestingly, I tried the nightly build again and it went for about 40 minutes before crashing for a second time. Since installing the nightly build, it crashes more easily, (before I could go 5 to 7 hours before a crash if lucky) . I normally continuously display a Zoom SCENE windowed projector and a windowed PREVIEW projector side-by-side. Prior to installing the nightly build, the Zoom SCENE would freeze first but  the PREVIEW project wouldn't freeze until the entire application hung. But now with the two crashes from the nightly build, (this crash and the above-reported crash), BOTH of the projectors freeze...

This last crash produced a crash log which is included below.
[Crash 2021-02-14 21-20-13.txt](https://github.com/obsproject/obs-studio/files/5979642/Crash.2021-02-14.21-20-13.txt)

The log file is here:
[2021-02-14 20-32-30.txt](https://github.com/obsproject/obs-studio/files/5979646/2021-02-14.20-32-30.txt)

**Comment by jpark37** [comment]
That PR may not fix your issue, but it should address the immediate crash.

**Comment by WizardCM** [comment]
@lcalder Please try again [with this nightly](https://obsbot.rodney.io/artifacts/41048119). It should no longer crash, but a session log will provide more info about what's going wrong.

**Comment by lcalder** [comment]
Thanks for the new nightly. I will try and provide you with the logs.

**Comment by lcalder** [comment]
I downloaded the nightly build recommended just above and ran for several hours without any crashes. When I tried to exit OBS it hung and had to be killed by right-clicking and then clicking on close program, and then close all windows.

The generated log file is attached here. I will try again later and include more log files. Thanks for all the help.
[2021-02-15 09-55-46.txt](https://github.com/obsproject/obs-studio/files/5984128/2021-02-15.09-55-46.txt)
